spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisAutoConfiguration.java

```diff
@@ -91,10 +98,25 @@ public JedisConnectionFactory redisConnectionFactory()
 
 		protected final JedisConnectionFactory applyProperties(
 				JedisConnectionFactory factory) {
-			factory.setHostName(this.properties.getHost());
-			factory.setPort(this.properties.getPort());
-			if (this.properties.getPassword() != null) {
-				factory.setPassword(this.properties.getPassword());
+			if (StringUtils.hasText(this.properties.getUrl())) {
+				try {
+					URI redisURI = new URI(this.properties.getUrl());
+					factory.setHostName(redisURI.getHost());
```

Comment:
SSL has no property in `RedisProperties` yet. You can set whether to use SSL with [`JedisConnectionFactory.setUseSsl(…)`](http://docs.spring.io/spring-data/data-redis/docs/1.8.0.M1/api/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.html#setUseSsl-boolean-).